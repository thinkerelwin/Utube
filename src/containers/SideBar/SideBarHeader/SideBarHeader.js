import React from 'react';
import './SideBarHeader.scss';
import PropTypes from 'prop-types';
import { Menu } from 'semantic-ui-react';

export function SideBarHeader(props) {
  const heading = props.title ? props.title.toUpperCase() : '';
  return (
    <Menu.Item>
      <Menu.Header className="side-bar-header">{heading}</Menu.Header>
    </Menu.Item>
  );
}

SideBarHeader.propTypes = {
  title: PropTypes.string,
};
