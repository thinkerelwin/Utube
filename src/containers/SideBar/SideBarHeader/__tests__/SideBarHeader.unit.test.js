import React from 'react';
import { shallow } from 'enzyme';
import { SideBarHeader } from '../SideBarHeader';

describe('SideBarHeader', () => {
  test('renders SideBarHeader with props.title=null', () => {
    const wrapper = shallow(<SideBarHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders SideBarHeader with props.title=''", () => {
    const wrapper = shallow(<SideBarHeader title="" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders SideBarHeader with sample title', () => {
    const wrapper = shallow(<SideBarHeader title="Sample Title" />);
    expect(wrapper).toMatchSnapshot();
  });
});
