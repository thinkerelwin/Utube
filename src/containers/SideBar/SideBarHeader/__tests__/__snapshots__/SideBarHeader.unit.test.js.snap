// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarHeader renders SideBarHeader with props.title='' 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  />
</MenuItem>
`;

exports[`SideBarHeader renders SideBarHeader with props.title=null 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  />
</MenuItem>
`;

exports[`SideBarHeader renders SideBarHeader with sample title 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  >
    SAMPLE TITLE
  </MenuHeader>
</MenuItem>
`;
